### DISCUSSION TITLE
How to Send idPrefix To Grid  When  Form Created

### DISCUSSION THREAD

**Discussion by eissa1005** [serenity-is/Serenity#5743] [Help]
/// <reference path="../../../Common/Helpers/SelectableEntityGrid.ts" />

namespace ALgorithmPro.ALgorithm {

    import FLD = ItemsRow.Fields;
    @Serenity.Decorators.registerClass()
    export class ItemsLookupGrid extends SelectableEntityGrid<ItemsRow, any> {
        protected getColumnsKey() { return 'ALgorithm.Items'; }
        protected getDialogType() { return ItemsDialog; }
        protected getIdProperty() { return ItemsRow.idProperty; }
        protected getInsertPermission() { return ItemsRow.insertPermission; }
        protected getLocalTextPrefix() { return ItemsRow.localTextPrefix; }
        protected getService() { return ItemsService.baseUrl; }

        protected form = new ASTRDForm("ALgorithmPro_ALgorithm_ASTRDDialog87_");
    
   }

   }
 }
===============
I Use IdPrefix But Constant  "ALgorithmPro_ALgorithm_ASTRDDialog87_"

**Comment by eissa1005** [comment]
@volkanceylan   Please Replay How Do this

**Reply by VictorTomaili** [reply]
Please don't tag people who doesn't wrote to your question. 

IdPrefix is autogenerated string what can accessed from widgets. You don't need pass this value to grid back except custom requirements. If you have custom requirements, you need to write this. Also It's just a string value, you can access dialog idPrefix from grid after dialog created on [this method](https://github.com/serenity-is/Serenity/blob/523c0195fec17089e1dd7e812650c2edaf47a7ac/src/Serenity.Scripts/CoreLib/UI/DataGrid/EntityGrid.ts#L304) after override it.

**Reply by eissa1005** [reply]
OK , I don't  use Tag  Again 
I Want Send Value When Select Row to this form but can't Create Form in this Grid 
namespace ALgorithmPro.ALgorithm {

    import FLD = ItemsRow.Fields;
    @Serenity.Decorators.registerClass()
    export class ItemsLookupGrid extends SelectableEntityGrid<ItemsRow, any> {
        protected getColumnsKey() { return 'ALgorithm.Items'; }
        protected getDialogType() { return ItemsDialog; }
        protected getIdProperty() { return ItemsRow.idProperty; }
        protected getInsertPermission() { return ItemsRow.insertPermission; }
        protected getLocalTextPrefix() { return ItemsRow.localTextPrefix; }
        protected getService() { return ItemsService.baseUrl; }


       protected form = new ASTRDForm("ALgorithmPro_ALgorithm_ASTRDDialog87_");

        constructor(container: JQuery) {
            super(container);

            var grid = this.slickGrid;
            
            var One: number = 1;
            grid.onSelectedRowsChanged.subscribe(() => {
                var selectedRow = grid.getSelectedRows()[0];
                var ItemsRow: ItemsRow = grid.getDataItem(selectedRow);

                if (ItemsRow != null || ItemsRow.ID > 0) {
                    this.form.Item_CD.value = ItemsRow.Item_CD;
                    this.form.ITM_NM_AR.value = ItemsRow.Item_Name_AR;
                    this.form.PKPRC.value = ItemsRow.PPRC6;
                    this.form.QTY.value = One;
                    this.form.Value.value = ItemsRow.PPRC6 * One;
                    this.form.PKCST.value = ItemsRow.UCOST;
                    this.form.ItemBAL.value = ItemsRow.ItemBAL;
                }

                this.element.closest(".ui-dialog").find(".ui-dialog-titlebar-close").click();

            });
        }

**Reply by eissa1005** [reply]
![dailog](https://user-images.githubusercontent.com/20481378/119722190-52cf6100-be6c-11eb-8075-af5475b7cd96.png)
![Grid](https://user-images.githubusercontent.com/20481378/119722203-5662e800-be6c-11eb-8c45-8cac9c0b2614.png)

**Comment by eissa1005** [comment]
@VictorTomaili 
 I run that code but return idPrefix  Relation this Form on Dialog but I call different form on this dialog

**Reply by VictorTomaili** [reply]
Create a property named "gridIdPrefix" on dialog and pass that value to your dialog like every other object.

**Reply by eissa1005** [reply]
I  create  Property It works with some screens  but I have a problem  when call Form Master on Detail Because it is called GridEditor First 
      public static HeaderPrefix: string;
        public static PrefixID: string;

        constructor() {
            super();

            
            this.ACCNOGrid = this.byId('ACCMFGrid');
            
            this.form = new CashPurchASTRHForm(this.idPrefix);
            CashPurchASTRHDialog.PrefixID = this.form.idPrefix;
            CashPurchASTRHDialog.HeaderPrefix = this.form.idPrefix;

            this.form.TR_NO.changeSelect2(() => {
                this.getNextNumber();
            });
=============================
 protected initDialog() {
            super.initDialog();
            CashPurchASTRHDialog.PrefixID = this.form.idPrefix;
        }
